Sources/SourceKitLSP/Swift/SwiftTestingScanner.swift

```diff
@@ -0,0 +1,378 @@
+}
+
+/// Contains information about a testing attribute such as `@Test` or `@Suite`.
+struct TestingAttributeData {
+  /// The display name in the attribute, if any.
+  let displayName: String?
+
+  /// The tags applied to the attribute.
+  let tags: [String]
+
+  /// Whether or not the attributed test is unconditionally disabled.
+  ///
+  /// Disabled tests can be presented differently in IDEs to indicate that they will never run, although they will still
+  /// be represented in the results.
+  let isDisabled: Bool
+
+  /// Whether or not the attributed test is hidden.
+  ///
+  /// Hidden tests are not reported by SourceKit-LSP and are not run automatically.
+  let isHidden: Bool
+
+  /// Extract the testing attribute data from the given attribute, which is assumed to be an `@Test` or `@Suite`
+  /// attribute.
+  init(attribute: AttributeSyntax) {
+    // If the first argument is an unlabelled string literal, it is the
+    // display name of the test. Otherwise, the test does not have a display
+    // name.
+    if case .argumentList(let argumentList) = attribute.arguments,
+      let firstArgument = argumentList.first,
+      firstArgument.label == nil,
+      let stringLiteral = firstArgument.expression.as(StringLiteralExprSyntax.self)
+    {
+      self.displayName = stringLiteral.representedLiteralValue
+    } else {
+      self.displayName = nil
+    }
+
+    let traitArguments = traits(ofTestAttribute: attribute)
+
+    // Map the arguments to tag's names.
+    self.tags = traitArguments.lazy
+      .compactMap { $0.as(FunctionCallExprSyntax.self) }
+      .filter { functionCall in
+        switch functionCall.calledExpression.as(MemberAccessExprSyntax.self)?.fullyQualifiedName {
+        case "tags", "Tag.List.tags", "Testing.Tag.List.tags":
+          return true
+        default:
+          return false
+        }
+      }.flatMap(\.arguments)
+      .compactMap { $0.expression.as(StringLiteralExprSyntax.self)?.representedLiteralValue }
+
+    self.isDisabled = traitArguments.lazy
+      .compactMap { $0.as(FunctionCallExprSyntax.self) }
+      .filter { functionCall in
+        switch functionCall.calledExpression.as(MemberAccessExprSyntax.self)?.fullyQualifiedName {
+        case "disabled", "ConditionTrait.disabled", "Testing.ConditionTrait.disabled":
+          return true
+        default:
+          return false
+        }
+      }
+      .contains { functionCall in
+        // Ignore disabled traits which have an `if:` parameter since
+        // they're conditional.
+        let hasConditionParam = functionCall.arguments.lazy
+          .compactMap(\.label?.text)
+          .contains("if")
+        if hasConditionParam {
+          return false
+        }
+
+        // Ignore disabled traits which have a trailing closure since
+        // they're conditional.
+        if functionCall.trailingClosure != nil {
+          return false
+        }
+
+        return true
+      }
+
+    self.isHidden = traitArguments.lazy
```

Comment:
We probably don't need to scan for `.hidden`: it's an internal-only trait we use for our test fixtures, not one external developers can use.

Reply:
If we decide to not support `.hidden`, I’ll change it in a follow-up PR to unblock https://github.com/apple/sourcekit-lsp/pull/1175